sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -208,8 +209,12 @@ public int hashCode() {
     public static final Set<TypeName> STRING_TYPES = ImmutableSet.of(STRING);
     public static final Set<TypeName> DATE_TYPES = ImmutableSet.of(DATETIME);
     public static final Set<TypeName> CONTAINER_TYPES = ImmutableSet.of(ARRAY);
+    public static final Set<TypeName> MAP_TYPES = ImmutableSet.of(MAP);
     public static final Set<TypeName> COMPOSITE_TYPES = ImmutableSet.of(ROW);
 
+    public boolean isPrimitiveType() {
+      return isNumericType() || isStringType() || isDateType();
```

Comment:
<!--new_thread; commit:78f845e1215b15d3246bedd7985d4c2b737650b1; resolved:0-->
this is not correct (e.g. it excludes boolean). better off making this exclusive (return !isContainterType() && !isCompositeType()).

Reply:
right, will change